syntax = "proto3";

package blog;

service PostService {
  rpc ListPosts (Empty) returns (PostListReply) {}
}

message Empty {}

message Autor {
  string nome = 1;
  string email = 2;
  repeated string Posts = 3;
}

message Comment {
  string autorDoComentario = 1;
  string conteudo = 2;
}

message Post {
  string titulo = 1;
  Autor autor = 2;
  repeated Comment comments = 3;
}

message PostListReply {
  repeated Post posts = 1;
}